executor.go

```diff
@@ -92,28 +143,20 @@ func (e *executor) start() {
 					}()
 				}
 
-				if e.maxRunningJobs != nil {
-					if !e.maxRunningJobs.TryAcquire(1) {
-
-						switch e.limitMode {
-						case RescheduleMode:
-							return
-						case WaitMode:
-							select {
-							case <-e.ctx.Done():
-								return
-							case <-f.ctx.Done():
-								return
-							default:
-							}
-
-							if err := e.maxRunningJobs.Acquire(f.ctx, 1); err != nil {
-								break
-							}
+				if e.limitModeMaxRunningJobs > 0 {
+					switch e.limitMode {
+					case RescheduleMode:
+						e.limitModeQueueMu.Lock()
+						if e.limitModeQueue == nil || len(e.limitModeQueue) < e.limitModeMaxRunningJobs {
+							e.limitModeQueue = append(e.limitModeQueue, f)
 						}
+						e.limitModeQueueMu.Unlock()
+					case WaitMode:
+						e.limitModeQueueMu.Lock()
+						e.limitModeQueue = append(e.limitModeQueue, f)
+						e.limitModeQueueMu.Unlock()
 					}
-
-					defer e.maxRunningJobs.Release(1)
+					return
```

Comment:
Something with the starting and stopping resulted in a negative wait group counter on line 177 - needs some investigation